### ISSUE TITLE
Update to serve 0.4

### ISSUE THREAD

**Issue by msaroufim** [cceyda/torchserve-dashboard#1]
I love your project and was hoping we can feature it more prominently in the main torchserve repo - I was wondering if you'd be OK and interested in this. And if so I was wondering if you could give me some feedback on the below

## Installation instructions
I tried to `torchserve-dashboard --server.port 8105 -- --config_path ./torchserve.properties --model_store ./model_store` but the page never seems to load regardless of whether I use the network or external url that I have

I setup a config

```
(torchservedashboard) ubuntu@ip-172-31-1-60:~/torchserve-dashboard$ cat torchserve.properties 
inference_address=http://127.0.0.1:8443
management_address=http://127.0.0.1:8444
metrics_address=http://127.0.0.1:8445
grpc_inference_port=7070
grpc_management_port=7071
number_of_gpu=0
batch_size=1
model_store=model_store
```

But perhaps makes the most sense to just add a default one to the repo so things just work. I'm happy to make the PR just let me know what you suggest. Ideally things just work with zero config and people can come back and change stuff once they feel more comfortable. 

Also on Ubuntu I had to type `export PATH="$HOME/.local/bin:$PATH"` so I could call `torchserve-dashboard`

## Features
Also there's some new features we're excited like the below which would be very interesting to see like
1. Model interpretability with Captum https://github.com/pytorch/serve/blob/master/captum/Captum_visualization_for_bert.ipynb
2. Workflow support coming in 0.4 which will allow much more configurable pipelines https://github.com/pytorch/serve/pull/1024/files

In all cases please let me know if you think we're on the right track and how we can make the `torchserve` more useful to you. I liked your suggestion on automatic doc generation and it's something I'm looking into so please keep them coming!

**Comment by cceyda** [comment]
> I love your project and was hoping we can feature it more prominently in the main torchserve repo - I was wondering if you'd be OK and interested in this. 

Thank you~ sure I would be interested. 
So far I have been able to update this only whenever I do server/model maintenance and realize something 😬 

# Bug fix
The dashboard's visibility itself is only dependent on streamlit, which changes versions very quickly so that was probably the issue with the dashboard not showing up at all.
I pinned the version now `streamlit ==0.81.1` so there won't be sudden breaks 🤞 

> Also on Ubuntu I had to type export PATH="$HOME/.local/bin:$PATH" so I could call torchserve-dashboard

I noticed this too while installing on other machines. I'm quiet new to the whole python packaging world 😅 so I don't really know how to fix this. using `pip install --user ` seems to do the trick (sometimes). I will look into it later~

>  Ideally things just work with zero config and people can come back and change stuff once they feel more comfortable.

Agree, I have changed it to use a default config if not provided & update the readme. There is a now a new release on pypi~

Also added some more checks and error messages

# My todo's
I plan on refactoring this sometime;

- remove streamlit dependency from api. So that api.py will be an independent python wrapper around the inference http endpoints. (Will make maintenance easier!)
-  make it possible to also manipulate a torchserve instance by just inputting the management address (on an already started torchserver). So that dash.py will have 2 modes local and remote:
    - local: the current version that starts/stops a local torchserve instance.
    - remote: takes management address (accessible by where the dashboard is running). can remove/scale/get info on models. register model (paths should be available)

# Features
- [Workflows](https://github.com/pytorch/serve/issues/682) look interesting. Will watch the PR to reflect changes to the API

**Comment by FlorianMF** [comment]
Hey, I'm a big fan of this UI of Torchserve. This is definitely a missing piece atm.
To show work progress to stakeholders I added a InferenceAPI class to the existing code.
I added the options to either call the predictions or the explanations endpoint.
atm I'm trying to make it work with different tasks for computer vision (classification, detection, keypoints) and adjust the shown result based on the response type (dictionary vs image)

**Comment by cceyda** [comment]
@FlorianMF Thank you 😄 If you would like to open a PR here or have a repo, I can help you out with it.
You might also be interested in https://github.com/pytorch/serve/issues/1043
& this
![image](https://user-images.githubusercontent.com/15624271/122543555-dc86de80-d066-11eb-88cb-2e1e8706be29.png)

I think It would be very nice to have an InferenceAPI!

**Comment by cceyda** [comment]
@FlorianMF  Looking at https://pytorch.org/serve/inference_api.html#predictions-api & https://github.com/pytorch/serve/blob/fd4e3e8b72bed67c1e83141265157eed975fec95/frontend/server/src/test/resources/inference_open_api.json#L146
PredictionsAPI (aka InferenceAPI) doesn't (yet?) generate a detailed enough schema for the requestBody to generate reasonable UI components for it. Other than a possible OpenAPI spec we don't have a generalized/supported way to know whether a model takes a string/image/etc as input body.
I guess we will have to wait until there is some movement on https://github.com/pytorch/serve/issues/1043

**Comment by cceyda** [comment]
I opened issues above to track the specific features mentioned here.
Since I already updated & released something for serve 0.4 I will close this issue.
